File: lab1-22.02.21/src/main/java/testpassword/lab1/repos/UserRepo.java
package testpassword.lab1.repos;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import testpassword.lab1.models.User;

@Repository public interface UserRepo extends CrudRepository<User, Long> {

    User getByEmail(String email);

    User getByUserId(long id);

    boolean existsByEmail(String email);
}

File: lab3-18.05.21/src/main/kotlin/lab3/repos/UserRepo.kt
package lab3.repos

import lab3.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository interface UserRepo: JpaRepository<User, Long> {

    infix fun getByEmail(email: String): User

    infix fun getByUserId(id: Long): User
}